src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisitionOptions.cs

```diff
@@ -25,6 +26,47 @@ public class TokenAcquisitionOptions : AcquireTokenOptions
         /// </summary>
         public CancellationToken CancellationToken { get; set; } = CancellationToken.None;
 
+        /// <summary>
+        /// Creates a new instance of TokenAcquisitionOptions from an instance of the base class AcquireTokenOptions.
+        /// </summary>
+        /// <param name="acquireTokenOptions">The instance of the base class to clone</param>
+        /// <returns>A new instance of TokenAquisitionOptions</returns>
+        public static TokenAcquisitionOptions CloneFromBaseClass(AcquireTokenOptions? acquireTokenOptions)
+        {
+            if (acquireTokenOptions == null)
+            {
+                return new TokenAcquisitionOptions();
+            }
+
+            // alternate option 1: serialize the base class then deserialize it back to the derived class, this is probably not
+            // ideal since it likely has small performance repercussions but it doesn't have to be done with json.
+
+            // alternate option 2: add the two extra fields of PoPConfiguration and CancellationToken to the base class in
+            // Abstractions and leave this child as a 1:1 wrapper with the base class to avoid breaking anyone.
+            // This is my personal preference as it is most performant since no conversion would be needed. The drawback is I'd
+            // need to replace all instances of TokenAcquisitionOptions with AcquireTokenOptions in the codebase.
+            //      2.1 - I spoke with JM and he said this would be a longer conversation as he wants to deprecate
+            //      TokenAcquisitionOptions entirely and drop the CancellationToken and PoPConfiguration fields as they
+            //      apparently don't belong in AcquireTokenOptions.
+
+            // This is how we have been doing it so far (in the DefaultAuthorizationHeaderProvider class) I put it here to make
+            // it easier to find if we ever add more fields to the base class, but it will still require manual intervention to
+            // add the new fields here.
+            return new TokenAcquisitionOptions
```

Comment:
Not quite right, and agree that we will deprecate these options at some point (hopefully soon for .NET 9)

Why not just do `ManagedIdentity = downstreamApiOptions?.AcquireTokenOptions.ManagedIdentity,` with the others?

Reply:
See above comment, TLDR: it was a design decision to put the logic with the object it relates to rather than a random other class but if you'd prefer I put it back I can do that.